src/common/SIRF.py

```diff
@@ -98,7 +99,10 @@ def dot(self, other):
         data viewed as vectors.
         other: DataContainer
         '''
-        assert_validities(self, other)
+        assert_validities(self,other)
+        # Check if input are the same size
+        if numpy.prod(self.dimensions()) != numpy.prod(other.dimensions()):
+            raise ValueError("Input sizes are expected to be equal, got " + numpy.prod(self.dimensions()) + " and " + numpy.prod(other.dimensions()) + " instead.")
```

Comment:
should it be moved to `assert_validities`? Not sure where that function is called, but it seems to me that a size check should be useful in many places.

Reply:
Can't answer this now! I can try to check later when is `assert_validities` used in the entire SIRF and confirm if it fits there.

Reply:
`assert_validities` is a type check, afaik. I don't know if we always want to be sure that the type and size of the object is correct. Possibly this could be a different function `assert_size_match` or whatever

Reply:
I am sure that `assert_vaidity` should check equal sizes.

Reply:
@evgueni-ovtchinnikov inside it goes then.